app/file-detail/template.hbs

```diff
@@ -1,29 +1,161 @@
+                                        <textarea readonly='true' class='form-control'>{{{shareiFrameDirect}}}</textarea>
+                                    {{/tab.pane}}
+                                </div>
+                            {{/bs-tab}}
+                        {{/bs-popover}}
+                    </button>
+                </div>
+                <div class='btn-group m-l-xs m-t-xs'>
+                    <button class='btn btn-sm btn-primary' {{action 'download' model.file.currentVersion}}>
+                        Download
+                    </button>
+                </div>
+                <div class='btn-group m-l-xs m-t-xs'>
+                    <button class='btn btn-sm btn-primary view-button' {{action 'changeView'}}>
+                        View
+                    </button>
+                </div>
+                <div class='btn-group m-l-xs m-t-xs'>
+                    <button class='btn btn-sm btn-default view-button' {{action 'changeView'}}>
+                        Revisions
+                    </button>
+                </div>
+            </div>
         </div>
     </div>
 </div>
 <hr>
 <div class='row'>
     <div class='col-md-3'>
         {{file-browser
-            selectedFile=model
+            selectedFile=model.file
             display=displays
-            userId=userId
+            user=model.user
             openFile=(action 'openFile')
         }}
+        <div class='panel panel-default'>
+            <div class='panel-heading clearfix'>
+                <h3 class='panel-title'>Tags:</h3>
+            </div>
+            <div class='panel-body'>
+                {{#if edit}}
+                    {{#tag-input
+                        tags=fileTags
+                        addTag=(action 'addTag')
+                        removeTagAtIndex=(action 'removeTagAtIndex')
+                        allowSpacesInTags=true
+                        as |tag|
+                    }}
+                        {{tag}}
+                    {{/tag-input}}
+                {{else}} <!-- Passing a mutable value to readOnly breaks tag-input. INstead render it with a set value -->
+                    {{#tag-input
+                        tags=fileTags
+                        addTag=(action 'addTag')
+                        removeTagAtIndex=(action 'removeTagAtIndex')
+                        allowSpacesInTags=true
+                        readOnly=true
+                        as |tag|
+                    }}
+                        {{tag}}
+                    {{/tag-input}}
+                {{/if}}
+            <div class="tags_clear">
+            </div>
+            </div>
+        </div>
     </div>
     <div class='col-md-9'>
-        {{#file-renderer download=model.links.download
-            width="800" height="1000"}}
-        {{/file-renderer}}
+        <div id='mfrIframeParent'>
+            {{#file-renderer download=model.file.links.download version=mfrVersion
+                height="1000"}}
```

Comment:
Not sure on this one. Do we need to pass in a set height? How does that work with a file that is smaller than 1000 height

Reply:
Haven't tried it on a smaller file, but it looks like it has a property that sets the width/height to 100% if nothing is passed through.  Maybe it would be better to remove the height of 1000 in this case.

Reply:
Checked how it worked for preprints.  It looks like it gives a height of 700px and a width of 99%.  For now, I'm changing the height and width to match what it is on preprints.